src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirHealthService.cs

```diff
@@ -0,0 +1,45 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Rest;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class R4FhirHealthService :
```

Comment:
Can we have unit tests with mocks of the exceptions to test the various outputs?

Reply:
Added tests